### PR TITLE
kubevirt's components managed-by label's value changed to virt-operator 

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:
Changed 'managed-by' label's value of kubevirt components to be 'virt-operator'
instead of 'kubevirt-operator', since the operator name is virt-operator.
we need it because the value 'kubevirt-operator' could be misleading.
A similar PR has been approved (#6565 ) and it caused failure on update( #6763 ),
This PR consider the update.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
To test if this PR handle update i simulated an updated from 0.46.1 to this branch and verified with `kubevirt logs <operator's name> -n kubevirt`
**Release note**:

```release-note
'kubevirt-operator' changed to 'virt-operator' on 'managed-by' label in kubevirt's components made by virt-operator
```

### CODE REVIEW

**Comment by Barakmor1** [comment]
@zcahana

File: pkg/controller/virtinformers.go
Comment (zcahana) [inline pkg/controller/virtinformers.go:69]: I'd take this chance to rename this misleadingly named variable, since it doesn't represent a label but rather a label selector. `OperatorLabelSelector`? Also recommend to use `fmt.Sprintf` here since it'll result with more readable (and less prone to whitespace/parentheses mistakes):
```suggestion
OperatorLabelSelector = fmt.Sprintf("%s in (%s,%s)", 
	kubev1.ManagedByLabel,
	kubev1.ManagedByLabelOperatorValue,
	kubev1.ManagedByLabelOperatorValueOld)
```

**Review by zcahana — CHANGES_REQUESTED** [review]
Nice work, I've made a couple of comments inline, and here are a couple more:

1. I found two more places where this label is referenced, we'd need to update them too to consider the update logic: [here](https://github.com/kubevirt/kubevirt/blob/e622b99f6cfe4cbd659ff408dffb411effa2664c/pkg/virt-operator/util/types.go#L85) and [here](https://github.com/kubevirt/kubevirt/blob/e622b99f6cfe4cbd659ff408dffb411effa2664c/pkg/virt-operator/resource/apply/reconcile.go#L70).
2. In general it would have been ideal to have all the logic regarding the ManagedBy label in the same place (label selector definition, `IsManagedByOperator` func, ..). The `pkg/virt-operator/util` could have been a good spot, but it imports `pkg/controller` so moving all that code in there would cause an import cycle. To avoid the cycle, we can create a new `pkg/virt-operator/util/label` pkg for that.
3. I think it would be good to have this PR tested against the new test in #6786, so it's worth to wait a few days to let it merge and then rebase on top.

File: pkg/controller/virtinformers.go
Comment (Barakmor1) [reply pkg/controller/virtinformers.go:69]: Hey Zvi,
Thanks for the note.
Actually i already tried to do that but it's not possible because this code is inside a `const{..}` block
and fmt.Sprintf doesn't return const value and in my opinion it's better to keep OperatorLabelSelector const
than make this code more readable.

Comment (zcahana) [reply pkg/controller/virtinformers.go:69]: You're right, I missed that. Converting to `var` is quite a common practice for defining runtime-evaluated "constants" (here, the constness of it is only by convention). Another option to achieve constness is by defining it as a function. Either way this isn't a big thing, and can be left as is too (I'd still rename the variable though ;-) )

File: pkg/controller/virtinformers.go
Comment (Barakmor1) [reply pkg/controller/virtinformers.go]: totally, done.

File: staging/src/kubevirt.io/client-go/apis/core/v1/types.go
Comment (Barakmor1) [reply staging/src/kubevirt.io/client-go/apis/core/v1/types.go:732]: done.

File: pkg/virt-operator/resource/apply/reconcile.go
Comment (zcahana) [inline pkg/virt-operator/resource/apply/reconcile.go:71]: how about just calling `util.IsManagedByOperator(objectMeta.Labels)`?

**Comment by Barakmor1** [comment]
retest

**Comment by zcahana** [comment]
Hi @Barakmor1, I can see the sig-operator lanes are failing with the new upgrade test from #6786 (the rest of the failures in this lanes are probably just cascades, so ignore them). The failing assertion is [here](https://github.com/kubevirt/kubevirt/blob/35966502dec73d075cc953099475a40310fa8672/tests/operator_test.go#L301) and it means that at the beginning of the test, there were no running/ready pods in the `kubevirt` namespace with the new managed-by label. Looking at the some more [test logs](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6796/pull-kubevirt-e2e-k8s-1.22-operator/1466312801961119744/artifacts/k8s-reporter/11_pods.log), it appears that indeed the kubevirt pods are labelled with the old `kubevirt-operator` labels.

I suggest to run this test (the whole test_id:3145 table) locally and continue debug from there.

**Comment by Barakmor1** [comment]
> Hi @Barakmor1, I can see the sig-operator lanes are failing with the new upgrade test from #6786 (the rest of the failures in this lanes are probably just cascades, so ignore them). The failing assertion is [here](https://github.com/kubevirt/kubevirt/blob/35966502dec73d075cc953099475a40310fa8672/tests/operator_test.go#L301) and it means that at the beginning of the test, there were no running/ready pods in the `kubevirt` namespace with the new managed-by label. Looking at the some more [test logs](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6796/pull-kubevirt-e2e-k8s-1.22-operator/1466312801961119744/artifacts/k8s-reporter/11_pods.log), it appears that indeed the kubevirt pods are labelled with the old `kubevirt-operator` labels.
> 
> I suggest to run this test (the whole test_id:3145 table) locally and continue debug from there.

Thanks for informing me ,
will do.

**Comment by Barakmor1** [comment]
/cc @zcahana 
Hey Zvi can you have a look?

**Comment by zcahana** [comment]
Looks good @Barakmor1, thanks!
/lgtm

Let's wait for a final ack from an approver.

**Review by stu-gott — COMMENTED** [review]
/approve

**Pull request merged**
